image-rs/src/stream/unpack.rs

```diff
@@ -25,18 +26,90 @@ use tokio::{fs, io::AsyncRead};
 use tokio_tar::ArchiveBuilder;
 use xattr::FileExt;
 
+pub type UnpackResult<T> = std::result::Result<T, UnpackError>;
+
+#[derive(Error, Debug)]
+pub enum UnpackError {
+    #[error("Failed to delete existed broken layer when unpacking")]
+    DeleteExistingLayerFailed {
+        #[source]
+        source: io::Error,
+    },
+
+    #[error("Create layer directory failed")]
+    CreateLayerDirectoryFailed {
+        #[source]
+        source: io::Error,
+    },
+
+    #[error("Failed to read entries from the tar")]
+    ReadTarEntriesFailed {
+        #[source]
+        source: io::Error,
+    },
+
+    #[error("Illegal entry name in the layer tar: {0}")]
+    IllegalEntryName(String),
+
+    #[error("Failed to get a legal UID of the file")]
+    IllegalUid {
+        #[source]
+        source: anyhow::Error,
+    },
+
+    #[error("Failed to get a legal GID of the file")]
+    IllegalGid {
+        #[source]
+        source: anyhow::Error,
+    },
+
+    #[error("Failed to get a legal mtime of the file")]
+    IllegalMtime {
+        #[source]
+        source: io::Error,
+    },
+
+    #[error("Convert whiteout file failed: {source}")]
+    ConvertWhiteoutFailed {
+        #[source]
+        source: anyhow::Error,
+    },
+
+    #[error("Failed to unpack layer to destination")]
+    UnpackFailed {
+        #[source]
+        source: io::Error,
+    },
+
+    #[error("Failed to set ownerships for path: {path}")]
```

Comment:
nit: 'ownerships` -> 'ownership`

Reply:
This is to align with the [function ](https://github.com/confidential-containers/guest-components/blob/main/image-rs/src/unpack.rs#L210-L248) where both UID and GID are both changed. This might be proper to name as `ownerships`?